object-storage-core/src/main/java/io/micronaut/objectstorage/UploadRequest.java

```diff
@@ -49,87 +54,107 @@ static UploadRequest fromBytes(@NonNull byte[] bytes,
     /**
      * @return the content type of this upload request.
      */
+    @NonNull
     Optional<String> getContentType();
 
     /**
      * @return the file name with path.
      */
+    @NonNull
     String getKey();
 
     /**
      * @return the size of the part, in bytes.
      */
+    @NonNull
     Optional<Long> getContentSize();
 
     /**
      * @return an input stream of the object to be stored.
      */
+    @NonNull
     InputStream getInputStream();
 
     /**
      * Upload request implementation using {@link java.io.File}.
      */
     class FileUploadRequest implements UploadRequest {
 
+        private static final Logger LOG = LoggerFactory.getLogger(FileUploadRequest.class);
+
         private final String keyName;
         private final String contentType;
         private final Path path;
 
-        public FileUploadRequest(Path localFilePath) {
+        public FileUploadRequest(@NonNull Path localFilePath) {
             this(localFilePath, localFilePath.getFileName().toString(), null,
                 URLConnection.guessContentTypeFromName(localFilePath.toFile().getName()));
         }
 
-        public FileUploadRequest(Path localFilePath, String prefix) {
+        public FileUploadRequest(@NonNull Path localFilePath,
+                                 @Nullable String prefix) {
             this(localFilePath, localFilePath.toFile().getName(), prefix,
                 URLConnection.guessContentTypeFromName(localFilePath.toFile().getName()));
         }
 
-        public FileUploadRequest(Path localFilePath,
-                                 String keyName,
+        public FileUploadRequest(@NonNull Path localFilePath,
+                                 @NonNull String keyName,
                                  @Nullable String prefix,
                                  @Nullable String contentType) {
             this.keyName = prefix != null ? prefix + "/" + keyName : keyName;
             this.contentType = contentType;
             this.path = localFilePath;
         }
 
+        @NonNull
         public File getFile() {
             return path.toFile();
         }
 
+        @NonNull
         public Path getPath() {
             return path;
         }
 
+        @NonNull
         public String getAbsolutePath() {
             return path.toAbsolutePath().toString();
         }
 
         @Override
+        @NonNull
         public Optional<String> getContentType() {
             return Optional.ofNullable(contentType);
         }
 
         @Override
+        @NonNull
         public String getKey() {
             return keyName;
         }
 
         @Override
+        @NonNull
         public Optional<Long> getContentSize() {
             try {
                 return Optional.of(Files.size(path));
             } catch (IOException e) {
+                if (LOG.isErrorEnabled()) {
```

Comment:
remove the 2 is-enabled checks, they're not needed for SLF4j when logging a simple string and just bulk up the code

Reply:
[I've removed the logging statement. ](https://github.com/micronaut-projects/micronaut-object-storage/pull/85#issuecomment-1231885433)